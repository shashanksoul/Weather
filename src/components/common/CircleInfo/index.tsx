import React from 'react';
import {Text, View} from 'react-native';

import style from './styles';

type CircleBaseProps = {
  temperature: string;
  isFeelsLike?: boolean;
};
type CircleExtendedProps = CircleBaseProps & {
  isFeelsLike: false;
  low: string;
  high: string;
};

type CircleInfoProps = CircleBaseProps | CircleExtendedProps;

const CircleInfo: React.FC<CircleInfoProps> = props => {
  if (props.isFeelsLike) {
    return (
      <View testID="circle-view" style={style.container}>
        {props.isFeelsLike && (
          <Text testID="feel-like-text" style={style.label}>
            FEELS LIKE
          </Text>
        )}
        <Text testID="temperature-text" style={style.temp}>
          {`${props.temperature}ยบ`}
        </Text>
      </View>
    );
  } else {
    const {low, high} = props as CircleExtendedProps;
    return (
      <View testID="circle-view" style={style.container}>
        <Text testID="temperature-text" style={style.temp}>
          {`${props.temperature}ยบ`}
        </Text>
        <Text
          testID="hi-low-text"
          style={style.hiLowTemp}>{`H: ${high}ยบ L: ${low}ยบ`}</Text>
      </View>
    );
  }
};

export default CircleInfo;
