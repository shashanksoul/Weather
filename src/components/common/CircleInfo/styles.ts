import {StyleSheet} from 'react-native';

import {SCREEN_WIDTH} from 'src/utils/Dimensions';

const circleWidth = SCREEN_WIDTH / 2;

const style = StyleSheet.create({
  container: {
    padding: 20,
    backgroundColor: 'black',
    borderRadius: circleWidth / 1,
    width: circleWidth - 40,
    height: circleWidth - 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  label: {
    fontFamily: 'SpaceMono-Regular',
    fontSize: 14,
    color: 'white',
  },
  temp: {
    fontFamily: 'SpaceMono-Regular',
    fontSize: 40,
    color: 'white',
  },
  hiLowTemp: {
    fontSize: 14,
    color: 'white',
  },
});

export default style;
