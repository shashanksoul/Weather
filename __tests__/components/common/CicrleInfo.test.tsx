import React from 'react';

import CircleInfo from '@components/common/CircleInfo';
import {render} from '@testing-library/react-native';

describe('CircleInfo Component', () => {
  test('render- ifFeelsLike true', () => {
    const circleInfo = render(<CircleInfo temperature="28" isFeelsLike />);
    const text = circleInfo.getByTestId('feel-like-text');
    expect(text).toBeTruthy();
    expect(circleInfo.toJSON()).toMatchSnapshot();
  });
  test('render- ifFeelsLike false', () => {
    const circleInfo = render(
      <CircleInfo temperature="28" high="29" low="18" />,
    );
    const tempText = circleInfo.getByTestId('temperature-text');
    const highLowText = circleInfo.getByTestId('hi-low-text');
    expect(tempText.children).toEqual(['28ยบ']);
    expect(highLowText.children).toEqual(['H: 29ยบ L: 18ยบ']);
  });
});
