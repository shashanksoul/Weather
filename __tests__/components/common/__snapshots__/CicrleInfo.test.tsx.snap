// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleInfo Component render- ifFeelsLike true 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 375,
      "height": 335,
      "justifyContent": "center",
      "padding": 20,
      "width": 335,
    }
  }
  testID="circle-view"
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "SpaceMono-Regular",
        "fontSize": 14,
      }
    }
    testID="feel-like-text"
  >
    FEELS LIKE
  </Text>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "SpaceMono-Regular",
        "fontSize": 40,
      }
    }
    testID="temperature-text"
  >
    28ยบ
  </Text>
</View>
`;
